File: 2022/august/15-08-2022/roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        values = {
            "I": 1,
            "V": 5,
            "X": 10,
            "L": 50,
            "C": 100,
            "D": 500,
            "M": 1000,
        }

        last = None
        res = 0
        for curr in s:
            if last and values[curr] > values[last]:
                res -= 2 * values[last]
            res += values[curr]
            last = curr
        return res

File: 2022/august/15-08-2022/roman-to-integer.rs
impl Solution {
    pub fn roman_to_int(s: String) -> i32 {
        let (mut res, mut prev, mut curr) = (0, 0, 0);
        for i in s.chars().rev() {
            match i {
                'I' => curr = 1,
                'V' => curr = 5,
                'X' => curr = 10,
                'L' => curr = 50,
                'C' => curr = 100,
                'D' => curr = 500,
                'M' => curr = 1000,
                _ => return -1,
            };
            if curr < prev { res = res - curr } else { res = res + curr };
            prev = curr;
        }
        res
    }
}